components/AppServices/CommunityToolkit.AppServices.SourceGenerators/AppServiceGenerator.cs

```diff
@@ -91,11 +98,18 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
                         return default;
                     }
 
+                    token.ThrowIfCancellationRequested();
+
                     INamedTypeSymbol typeSymbol = (INamedTypeSymbol)context.TargetSymbol;
 
                     // Get the info on the host implementation
                     HierarchyInfo hierarchy = HierarchyInfo.From(typeSymbol, typeSymbol.Name.Substring(1));
-                    ImmutableArray<MethodInfo> methods = MethodInfo.From(typeSymbol);
+
+                    token.ThrowIfCancellationRequested();
+
+                    ImmutableArray<MethodInfo> methods = MethodInfo.From(typeSymbol, token);
+
+                    token.ThrowIfCancellationRequested();
```

Comment:
Are these calls to ThrowIfCancellationRequested *too* often than needed?
I really have no idea, but it would be good to see if you can discuss with Roslyn team

Reply:
I'm generally adding them between macro-blocks, same as in the MVVM Toolkit. FWIW, checking for cancellation is extremely fast, so there isn't really a concern if you even check too often (it's mostly just a dereference and a couple branches). That said, not sure whether there's a specific guidance on exactly how often to check 🤔

Yeah, might be interesting to have a conversation about this, for sure 🙂

Reply:
To keep things moving, let's move this to a new Issue for discussion.